From e2b4f4c92b7923945c7789f13e850098ee20984b Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:50 -0400
Subject: [ethernet] bnx2x: Fix the MSI flags

Message-id: <1409946615-5862-46-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91586
O-Subject: [RHEL7.1 PATCH 45/70] bnx2x: Fix the MSI flags
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit 73413ffac3b713231dac466bca216f970042c5e5
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Wed Jun 25 12:22:56 2014 +0800

    bnx2x: Fix the MSI flags

    MSI-X should use PCI_MSIX_FLAGS not PCI_MSI_FLAGS.

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
index f86677f..7e78ae4 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
@@ -12946,7 +12946,7 @@ static int bnx2x_get_num_non_def_sbs(struct pci_dev *pdev, int cnic_cnt)
   * without the default SB.
   * For VFs there is no default SB, then we return (index+1).
   */
- pci_read_config_word(pdev, pdev->msix_cap + PCI_MSI_FLAGS, &control);
+ pci_read_config_word(pdev, pdev->msix_cap + PCI_MSIX_FLAGS, &control);
 
  index = control & PCI_MSIX_FLAGS_QSIZE;
 
-- 
1.7.1